src/mesh/voxel_cloud.rs

```diff
@@ -685,6 +705,100 @@ impl ScalarField {
         }
         self.shrink_to_fit(volume_value_range_self)
     }
+    /// Computes interpolated values of two scalar fields. Prior to
+    /// interpolation, the scalar fields will be recomputed to distance fields.
+    /// The current scalar field will be mutated and resized to encompass values
+    /// from both scalar fields. The two scalar fields do not have to contain
+    /// voxels of the same size.  The interpolation is linear.
+    ///
+    /// `interpolation_factor` is a value between 0.0 and 1.0 defining the ratio
+    /// between the current value to the other value.
+    ///
+    /// The `volume_value_range` is an interval defining which values of the
+    /// scalar field should be considered to be a volume. The
+    /// `ScalarField::from_mesh` generates a scalar field, which marks volume
+    /// voxels with value `0`. `compute_distance_field` marks each voxel with a
+    /// value representing the voxel's distance from the original volume,
+    /// therefore the voxels right at the shell of the volume are marked 0, the
+    /// layer around them is marked 1 or -1 (inside closed volumes) etc. Once
+    /// the scalar field is populated with meaningful values, it is possible to
+    /// treat (perform boolean operations or materialize into mesh) on various
+    /// numerical ranges. Such range is specified ad-hoc by parameter
+    /// `volume_value_range`.
+    ///
+    /// # Panics
+    /// Panics if one of the volume value ranges is infinite.
+    ///
+    /// # Warning
+    /// If the input scalar fields are far apart, the resulting scalar field may
+    /// be huge.
+    pub fn interpolated_union_of_distance_fields<U>(
+        &mut self,
+        volume_value_range_self: &U,
+        other: &ScalarField,
+        volume_value_range_other: &U,
+        interpolation_factor: f32,
+    ) where
+        U: RangeBounds<f32>,
+    {
+        let mut other_clone = other.clone();
+        let bounding_box_self = self.bounding_box_cartesian();
+        let bounding_box_other = other_clone.bounding_box_cartesian();
+
+        if let Some(bounding_box) = BoundingBox::union(vec![bounding_box_self, bounding_box_other])
+        {
+            self.resize_to_cartesian_bounding_box(&bounding_box);
+            other_clone.resize_to_cartesian_bounding_box(&bounding_box);
+            self.compute_distance_field(volume_value_range_self);
+            other_clone.compute_distance_field(volume_value_range_other);
+
+            self.interpolate_to(&other_clone, interpolation_factor);
+        } else {
+            // Wipe the current scalar field if none of the scalar fields
+            // contained any volume voxels.
+            self.wipe();
+        }
```

Comment:
This is a bounding box union, meaning it should never happen if you provide two bounding boxes. Instead I'd `expect` here.

Reply:
it can! the values are being interpolated, so you can't know if the resulting number fall into the meshing range.

Reply:
True, but the condition is on the result of `BoundingBox::union` of two cartesian bounding boxes. IIRC those boxes are just computed from the dimensions of the scalar field without looking at voxel values. Maybe that wasn't supposed to be like that?

Reply:
you're right. this can't be an empty bounding box, unless both of the input bounding boxes are empty. this is rather an UI operation error

Reply:
Yeah, but empty bounding box is still not `None`.